SDCARD scripts refactoring

tiopex opened the pull request:
- update uboot and kernel to latest
- remove console specific variants modules.custom.sh, boot.cmd
- fix gmenu2x backlight
- remove poweroff in firstboot
- add possibility to load/unload tvout driver

tiopex committed:
SDCARD scripts refactoring

- update uboot and kernel to latest
- remove console specific variants modules.custom.sh, boot.cmd
- fix gmenu2x backlight
- remove poweroff in firstboot
- add possibility to load/unload tvout driver

Apaczer commented on main/apps/tvon/tvout-on.sh:
```diff
@@ -0,0 +1,7 @@
+touch /mnt/tvout
+killall -9 main
+sync
+
```
doesn't seem to work without ``reboot``

Apaczer replied:
also how do you plan to switch between NTSC/PAL ?

tiopex replied:
works without reboot with gmenux from source, or from you. Gmenux from 1.3.3 crash after restart

Apaczer replied:
oh, so we need to upload latest gmenu2x here (no need to use legacy 1.3.3, considering we are not even sure the src).

Apaczer commented:
https://github.com/MiyooCFW/sdcard/blob/0f8e06741bbbd5603ee9c8d97ef4b4d4e9b33a9b/etc/main#L100-L105

The elif condition on 103 line need also point to environment variable if there is no modules.custom.sh otherwise it will not init sleep command

### EDIT:
Done with https://github.com/MiyooCFW/sdcard/pull/31/commits/6953dcb9f5448793ffdba5c61fe7f30c7b3dc668

Apaczer commented on gen_boot_scr.sh:
```diff
@@ -1,11 +1,4 @@
-mkimage -C none -A arm -T script -d boot-bb2x.cmd boot/variants/bittboy2x_v2/boot.scr
-mkimage -C none -A arm -T script -d boot-bb35.cmd boot/variants/bittboy3.5/boot.scr
-mkimage -C none -A arm -T script -d boot-pocketgo-v90-q90.cmd boot/variants/pocketgo_TE/boot.scr
-mkimage -C none -A arm -T script -d boot-pocketgo-v90-q90.cmd boot/variants/v90_q90/boot.scr
```
maybe let us leave at least one (e.g. v90_q90) boot.scr configuration as example and move it to backup folder while genereting, if there arouse need for non-standard setenv combination

tiopex replied:
everything is in uboot

Apaczer replied:
Yes but it is s more convenient to do changes from sdcard repo with ``mkimage`` , otherwise you have to set SDK->change code in uboot->upload here to make it work as you intended (I will asure you that there are not many ppl who are comfy with that procedure).

At least we should add info to README how to write settings to boot.scr.

tiopex replied:
ok, I will add test-boot.cmd for testing purposes

Apaczer replied:
already done

Apaczer reviewed (changes requested):
other than that this works splendid! Nice job, no need for wacky firstboot procedure. Eventually we can delete all gmenu2x.conf files (they are autogenerated by gmenu2x itself on latest pingflood's code and were only needed for gmenunx on MiyooCFW current repo)

TriForceX commented:
btw, since we have the reboot-method working now, Is it better that we change the poweroff to reboot at the end of the firstboot?

https://github.com/MiyooCFW/sdcard/blob/0f8e06741bbbd5603ee9c8d97ef4b4d4e9b33a9b/boot/firstboot#L95-L96

tiopex commented:
there is no reboot or poweroff in firstboot

Apaczer commented:
We should think of a method to switch between r61520fb & st7789sfb drivers, it becomes more cumbersome without ``modules.custom.sh``. Previously I was using MODULES_CUSTOM entry in ``options.cfg`` as workaround which was defaulting to "r61520fb" when set to 0 - but with current refactoring it becomes useless.

Apaczer committed:
update gmenu2x & delete deprecated configs

- update gmenu2x from: https://github.com/Apaczer/gmenu2x/commit/7cbba2de7328acd0f14b9308e934a6f65ee9bfc1
- update main input.conf
- delete cfg: all is set auto. by gmenu2x at launch

Apaczer commented:
Now we're left only with ``boot-logo`` in variants, which we can change to smth common like "Miyoo 1'st gen" or "RetroGaming" instead of naming every console (we lack still XYC and SUP M3 in there). Afterwards we could dump whole /boot/variants directory idea.

TriForceX commented:
> Now we're left only with `boot-logo` in variants, which we can change to smth common like "Miyoo 1'st gen" or "RetroGaming" instead of naming every console (we lack still XYC and SUP M3 in there). Afterwards we could dump whole /boot/variants directory idea.

ive been thinking about this, or maybe just Miyoo or MiyooCFW

Apaczer committed:
gmenu2x cleanup + add TV applinks

- remove unused applinks
- remove deprecated clock values from link strings
- add ``tvoff`` & ``tvon`` links in apps section

TriForceX commented:
boot logo for generic consoles is ready https://github.com/MiyooCFW/logo/actions/runs/4329154568
![image](https://user-images.githubusercontent.com/16083854/222874489-b5dad7a8-af7a-40c9-aaa5-7fd1db4e7626.png)

tiopex commented:
> We should think of a method to switch between r61520fb & st7789sfb drivers, it becomes more cumbersome without `modules.custom.sh`. Previously I was using MODULES_CUSTOM entry in `options.cfg` as workaround which was defaulting to "r61520fb" when set to 0 - but with current refactoring it becomes useless.

Maybe add also script for v90 variants like for TV, to switch driver (new - without tearing, old - with tearing)?

Apaczer commented:
> boot logo for generic consoles is ready

tks TriForce (& @hi-ban for design) I'll do mayor clean-up tonight and we should be ready to merge this as 1'st part of refactoring

>Maybe add also script for v90 variants like for TV, to switch driver (new - without tearing, old - with tearing)?

yes that's an option until we implement it in frontend's code.

Apaczer committed:
remove redundant boot/variants dir

- rm all variants
- move daemon to /boot
- move boot-logo & update to Miyoo logo
- add generic scripts to /boot

Apaczer committed:
add random boot.cmd

Apaczer committed:
update main & firstboot script for "no variants" method

Apaczer committed:
redo st7789s fb initalization check

Apaczer committed:
minor fixes

main + modules_custom

Apaczer committed:
add TVMODE to ``options.cfg``

Apaczer committed:
a few corrections for ``daemon`` backend

- fixes daemon reading from script
- restore vol/backlight configs

Apaczer committed:
minor documenting

Apaczer committed:
add ``firsboot.custom.sh`` handling

by default disabled, but maybe usefull for flashing custom u-boot

Apaczer commented:
Ok it should be good, let's just wait for @TriForceX to update gmenunx repo. I have new *.yml ready for AUTO-BUILDS on main repo if this get merged.
